hazelcast/src/main/java/com/hazelcast/internal/networking/nio/NioThread.java

```diff
@@ -333,38 +333,19 @@ private boolean processTaskQueue() {
             if (task == null) {
                 break;
             }
-            executeTask(task);
+            task.run();
+            completedTaskCount.inc();
             tasksProcessed = true;
         }
         return tasksProcessed;
     }
 
-    private void executeTask(Runnable task) {
```

Comment:
The difficult part of making sure that the owner of this task is the executing thread, is moved to the NioPipeline/NioPipelineTask. So we can get rid of that logic here